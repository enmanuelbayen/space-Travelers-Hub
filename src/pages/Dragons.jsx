import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchDragons } from '../Redux/dragons/dragonsSlice';
import Dragon from '../components/Dragon';
import '../assests/rockets.css';

const Dragons = () => {
  const dispatch = useDispatch();
  const { dragons, status, error } = useSelector((state) => state.dragons);

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchDragons());
    }
  }, [dispatch, status]);

  return (
    <div className="rocket-container">
      {status === 'loading' && <div>Loading...</div>}
      {status === 'succeeded' && (
      <ul className="rocket-list flex">
        {dragons.map((dragon) => (
          <Dragon
            key={dragon.id}
            name={dragon.name}
            type={dragon.type}
            image={dragon.flickr_images[0]}
            reserved={dragon.reserved}
          />
        ))}
      </ul>
      )}
      {status === 'failed' && <div>{error}</div>}
    </div>
  );
};
export default Dragons;
