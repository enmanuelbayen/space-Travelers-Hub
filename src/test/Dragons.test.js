import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Dragons from '../pages/Dragons';

const mockStore = configureStore([]);

describe('Dragons Component', () => {
  it('should match the snapshot', () => {
    const store = mockStore({
      dragons: {
        dragons: [
          {
            id: '1', name: 'Dragon 1', type: 'Type 1', flickr_images: ['image1'], reserved: false,
          },
          {
            id: '2', name: 'Dragon 2', type: 'Type 2', flickr_images: ['image2'], reserved: true,
          },
        ],
        status: 'succeeded',
        error: null,
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <Dragons />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
