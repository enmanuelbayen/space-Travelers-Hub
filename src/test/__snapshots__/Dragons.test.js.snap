// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragons Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="rocket-container"
  >
    <ul
      class="rocket-list flex"
    >
      <li>
        <img
          alt="Dragon 1"
          class="rocket-img"
          src="image1"
        />
        <div
          class="rocket-textBox flex"
        >
          <h3
            class="rocket-name"
          >
            Dragon 1
          </h3>
          <p
            class="rocket-description"
          >
            Type 1
          </p>
          <button
            class="btn col-2 btn-primary"
            type="button"
          >
            Reserve Dragon
          </button>
        </div>
      </li>
      <li>
        <img
          alt="Dragon 2"
          class="rocket-img"
          src="image2"
        />
        <div
          class="rocket-textBox flex"
        >
          <h3
            class="rocket-name"
          >
            Dragon 2
          </h3>
          <div
            class="reserved--badge"
          >
            RESERVED
          </div>
          <p
            class="rocket-description"
          >
            Type 2
          </p>
          <button
            class="btn col-2 btn-outline-danger"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
